src/main/resources/emulator.h

```diff
@@ -1549,7 +1549,7 @@ class mem_t {
   }
   val_t get (val_t idx, int word) {
     if (/*!ispow2(d) &&*/ idx >= d)
-      return __rand_val(seedp) & (word == val_n_words(w) && val_n_word_bits(w) ? mask_val(w) : -1L);
+      return __rand_val(seedp) & (((word+1) == val_n_words(w) && val_n_word_bits(w)) ? mask_val(w) : -1L);
```

Comment:
I defer to your judgement. I'm not sure val_n_full_words is very clear to me what it means, but if that's already a thing then maybe that's what should be used?